Source/Xamarin/Prism.Forms/Services/DeviceService.cs

```diff
@@ -3,78 +3,105 @@
 
 namespace Prism.Services
 {
-    /// <summary>
-    /// A service that exposes device-specific information and actions
-    /// </summary>
-    public class DeviceService : IDeviceService
-    {
-        /// <summary>
-        /// Gets the kind of device that Xamarin.Forms is currently working on.
-        /// </summary>
-        public TargetIdiom Idiom
-        {
-            get { return Device.Idiom; }
-        }
+	/// <summary>
+	/// A service that exposes device-specific information and actions
+	/// </summary>
+	public class DeviceService : IDeviceService
+	{
+		/// <summary>
+		/// Gets the kind of device that Xamarin.Forms is currently working on.
+		/// </summary>
+		public TargetIdiom Idiom
+		{
+			get { return Device.Idiom; }
+		}
 
-        /// <summary>
-        /// Gets the Platform (OS) that Xamarin.Forms is working on.
-        /// </summary>
-        public TargetPlatform Platform
-        {
-            get { return Device.OS; }
-        }
+		/// <summary>
+		/// Gets the Platform (OS) that Xamarin.Forms is working on.
+		/// </summary>
+		[Obsolete("Use RuntimePlatform instead")]
+		public TargetPlatform Platform
+		{
+			get { return Device.OS; }
+		}
 
-        /// <summary>
-        /// Invokes an action on the device main UI thread.
-        /// </summary>
-        /// <param name="action">The Action to invoke</param>
-        public void BeginInvokeOnMainThread(Action action)
-        {
-            Device.BeginInvokeOnMainThread(action);
-        }
+		/// <summary>
+		/// Gets the Platform (OS) that Xamarin.Forms is working on.
+		/// </summary>
+		public RuntimePlatform RuntimePlatform
+		{
+			get
+			{
+				switch (Device.RuntimePlatform)
+				{
+					case Device.Android:
+						return RuntimePlatform.Android;
+					case Device.iOS:
+						return RuntimePlatform.iOS;
+					case Device.Windows:
+						return RuntimePlatform.Windows;
```

Comment:
True, however this would be a new feature for a dev to implement, so might as well implement in a way where it will just work when 2.3.5 is released